Add config flow to System Monitor (#30115)
home-assistant/home-assistant.github.io: source/_integrations/systemmonitor.markdown

ha_release: pre 0.7
ha_iot_class: Local Push
ha_domain: systemmonitor
ha_config_flow: true
ha_platforms:
  - sensor
ha_integration_type: integration
---

The System monitor integration allows you to monitor disk usage,
memory usage, CPU usage, and running processes. 

{% include integrations/config_flow.md %}

## Sensors

**All entities are disabled by default, you need to enable the entities that you wish to use.**

### Disks

- Disk free - (One per disk/mount point)
- Disk use - (One per disk/mount point)
- Disk use (percent) - (One per disk/mount point)

### Network

- IPv4 address - (One per network interface)
- IPv6 address - (One per network interface)
- Network in - (One per network interface)
- Network out - (One per network interface)
- Packets in - (One per network interface)
- Packets out - (One per network interface)
- Network throughput in - (One per network interface)
- Network throughput out - (One per network interface)

### Other

- Last boot
- Load (15m)
- Load (5m)

> - Load (1m)

Reviewer: I see that other sites use m in this context as well. But would it be wrong to use min instead?

Reviewer 2: The "m" means minute so it wouldn't be wrong. Currently the sensors are named "Load (15m)" etc. in core. We should probably be consistent.

https://github.com/home-assistant/core/blob/fbcb31b1030e51cab2aac95484c4008fab40fbe2/homeassistant/components/systemmonitor/sensor.py#L123

I suggest the renaming to be done separately.

> The `systemmonitor` sensor platform allows you to monitor disk usage,

Reviewer 2: ```suggestion
The System monitor integration allows you to monitor disk usage,
```

memory usage, CPU usage, and running processes. 

{% include integrations/config_flow.md %}

## Sensors

**All entities are disabled by default, you need to enable the entities that you wish to use.**

### Disks

- Disk free - (One per disk/mount point)
- Disk use - (One per disk/mount point)
- Disk use (percent) - (One per disk/mount point)

### Network

- IPv4 address - (One per network interface)
- IPv6 address - (One per network interface)
- Network in - (One per network interface)
- Network out - (One per network interface)
- Packets in - (One per network interface)
- Packets out - (One per network interface)
- Network throughput in - (One per network interface)
- Network throughput out - (One per network interface)

### Other

- Last boot
- Load (15m)

> - Load (1m)
> - Load (5m)

Reviewer 2: ```suggestion
- Load (5m)
- Load (1m)
```

- Memory free
- Memory use
- Memory use (percent)
- Processor use
- Processor temperature
- Swap free
- Swap use
- Swap use (percent)

### Add `process` sensor

> The `process` sensor needs to be configured by the config entry options. Go to **{% my integrations title="Settings > Devices & Services" %}**, select the **System Monitor** integration and click **Configure** to add or remove `process` sensors.

Reviewer 2: ```suggestion
The `process` sensor needs to be configured by the config entry options. Go to **{% my integrations title="Settings > Devices & Services" %}**, select the **System Monitor** integration and click **Configure** to select which `process` sensors should be created.
```
